hadoop-plugin-test/src/main/gradle/negative/triggerCheck.gradle

```diff
@@ -0,0 +1,84 @@
+buildscript {
+  repositories {
+    jcenter()
+  }
+  dependencies {
+    classpath files("${project.pluginTestDir}/hadoop-plugin-${project.version}.jar", "${project.pluginTestDir}/hadoop-plugin-${project.version}-SNAPSHOT.jar")
+    // Must manually specify snakeyaml and quartz-scheduler for testing, not required for users
+    classpath 'org.yaml:snakeyaml:1.18'
+    classpath 'org.quartz-scheduler:quartz:2.2.1'
+  }
+}
+
+apply plugin: com.linkedin.gradle.hadoop.HadoopPlugin
+
+// Negative cases for trigger creation
+
+hadoop {
+  buildPath "jobs/triggerCheck"
+  cleanPath false
+
+  generateYamlOutput true
+
+  workflow('triggerFail1') {
+    trigger('trigger1') {
+      // no maxWaitMins throws an ERROR
+      // no schedule throws an ERROR
+      daliDependency('dali-data-dep') {
+        // no dali dependency view throws an ERROR
+        // no dali dependency window throws an ERROR
+        // no dali dependency delay throws an ERROR
+        // no dali dependency unit throws an ERROR
+      }
+    }
+
+    trigger('trigger2') { // two defined triggers throws an ERROR
+      maxWaitMins 0 // maxWaitMins less than 1 throws an ERROR
+      schedule {
+        // no schedule value throws an ERROR
+      }
+      schedule { // two schedules throws an ERROR
+        value 'abcd' // invalid cron expression for value in schedule throws an ERROR
+      }
+    }
+
+    commandJob('shellBash') {
+      uses 'bash ./sample_script.sh'
+    }
+
+    targets 'shellBash'
+  }
+
+  workflow('triggerFail2') {
+    trigger('trigger3') {
```

Comment:
So Triggers are one of the objects that can be defined in any BaseNamedScope. The benefit of this is that you can define the trigger in one place (like a global location) then reference it by name it will be added to a workflow.

Because of this, if you create a trigger named (for example) 'trigger1' outside of a workflow, then add it to a workflow with the command
```
addTrigger('trigger')
```
inside of that workflow, referencing by name.

So it does need _some_ name. I guess it doesn't necessarily have to be `trigger3`, but that does make sense given the other trigger names :)

Reply:
got it, thanks